backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxListenerApi.scala

```diff
@@ -142,13 +142,19 @@ class VeloxListenerApi extends ListenerApi with Logging {
     }
 
     SparkDirectoryUtil.init(conf)
-    UDFResolver.resolveUdfConf(conf, isDriver = false)
     initialize(conf, isDriver = false)
   }
 
   override def onExecutorShutdown(): Unit = shutdown()
 
   private def initialize(conf: SparkConf, isDriver: Boolean): Unit = {
+    JniWorkspace.initializeDefault(
+      SparkDirectoryUtil.get
+        .namespace("jni")
+        .mkChildDirRandomly(UUID.randomUUID.toString)
+        .getAbsolutePath)
```

Comment:
Compared with the original code, one difference is making dir is outside the synchronized block, which can cause extra directory unnecessarily created. Maybe, we can pass the making dir function as argument to initializeDefault and let the dir be created inside initializeDefault.